front/src/common/RollingStockSelector/RollingStockCardDetail.tsx

```diff
@@ -1,35 +1,50 @@
 import Loader from 'common/Loader';
-import { get } from 'common/requests';
-import PropTypes from 'prop-types';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useDispatch } from 'react-redux';
 import { setFailure } from 'reducers/main';
+import { RollingStock, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { ApiError } from 'common/api/emptyApi';
 import RollingStock2Img from './RollingStock2Img';
 import RollingStockCurves from './RollingStockCurves';
 
-const ROLLING_STOCK_URL = '/editoast/rolling_stock/';
+type RollingStockCardDetailProps = {
+  id: number;
+  curvesComfortList?: string[];
+  setCurvesComfortList: (curvesComfortList: string[]) => void;
+};
 
-export default function RollingStockCardDetail(props) {
+export default function RollingStockCardDetail({
+  id,
+  curvesComfortList,
+  setCurvesComfortList,
+}: RollingStockCardDetailProps) {
   const dispatch = useDispatch();
-  const { id, curvesComfortList, setCurvesComfortList } = props;
-  const [data, setData] = useState();
   const { t } = useTranslation(['rollingstock']);
 
-  const mode2name = (mode) => (mode !== 'thermal' ? `${mode}V` : t('thermal'));
+  const mode2name = (mode: string) => (mode !== 'thermal' ? `${mode}V` : t('thermal'));
 
-  const listCurvesComfort = (curvesData) => {
-    const list = ['STANDARD'];
+  const listCurvesComfort = (curvesData: RollingStock['effort_curves']) => {
+    const comfortList = ['STANDARD'];
     Object.keys(curvesData.modes).forEach((mode) => {
       curvesData.modes[mode].curves.forEach((curve) => {
-        if (!list.includes(curve.cond.comfort)) list.push(curve.cond.comfort);
+        if (curve.cond?.comfort) {
+          if (!comfortList.includes(curve.cond.comfort)) comfortList.push(curve.cond.comfort);
+        }
       });
     });
-    return list;
+    return comfortList;
   };
 
-  const transformCurves = (curvesData) => {
-    const transformedCurves = {};
+  const transformCurves = (curvesData: RollingStock['effort_curves']) => {
```

Comment:
Can you modify the function so it only take the modes in argument ?
```suggestion
  const transformCurves = (rollingStockCurves: RollingStock['effort_curves']['modes']) => {
```

Reply:
done